src/Umbraco.Core/Migrations/Upgrade/V_8_0_0/DropDownPropertyEditorsMigration.cs

```diff
@@ -0,0 +1,167 @@
+                {
+                    Database.Update(propData);
+                }
+
+                var requiresCacheRebuild = false;
+                switch (dd.EditorAlias)
+                {
+                    case "Umbraco.DropDown":
+                        UpdateDataType(dd, config, false);
+                        break;
+                    case "Umbraco.DropdownlistPublishingKeys":
+                        UpdateDataType(dd, config, false);
+                        requiresCacheRebuild = true;
+                        break;
+                    case "Umbraco.DropDownMultiple":
+                        UpdateDataType(dd, config, false);
+                        break;
+                    case "Umbraco.DropdownlistMultiplePublishKeys":
+                        UpdateDataType(dd, config, true);
+                        requiresCacheRebuild = true;
+                        break;
+                }
+
+                if (requiresCacheRebuild)
+                {
+                    //TODO: How to force rebuild the cache?
+                }
+            }
+        }
+
+        private void UpdateDataType(DataTypeDto dataType, ValueListConfiguration config, bool isMultiple)
+        {
+            dataType.EditorAlias = Constants.PropertyEditors.Aliases.DropDownListFlexible;
+            var flexConfig = new
+            {
+                multiple = isMultiple,
+                items = config.Items
+            };
+            dataType.Configuration = JsonConvert.SerializeObject(flexConfig);
+            Database.Update(dataType);
+        }
+
+        private bool UpdatePropertyDataDto(PropertyDataDto propData, ValueListConfiguration config)
+        {
+            //Get the INT ids stored for this property/drop down
+            int[] ids = null;
+            if (!propData.VarcharValue.IsNullOrWhiteSpace())
+            {
+                ids = ConvertStringValues(propData.VarcharValue);
+            }
+            else if (!propData.TextValue.IsNullOrWhiteSpace())
+            {
+                ids = ConvertStringValues(propData.TextValue);
+            }
+            else if (propData.IntegerValue.HasValue)
+            {
+                ids = new[] { propData.IntegerValue.Value };
+            }
+
+            //if there are INT ids, convert them to values based on the configured pre-values
+            if (ids != null && ids.Length > 0)
+            {
+                //map the ids to values
+                var vals = new List<string>();
+                var canConvert = true;
+                foreach (var id in ids)
+                {
+                    var val = config.Items.FirstOrDefault(x => x.Id == id);
+                    if (val != null)
+                        vals.Add(val.Value);
+                    else
+                    {
+                        Logger.Warn<DropDownPropertyEditorsMigration>($"Could not find associated data type configuration for stored Id {id}");
```

Comment:
Please don't do string interpolation of logging messages - otherwise, the Serilog template is not unique

```suggestion
                        Logger.Warn<DropDownPropertyEditorsMigration>("Could not find associated data type configuration for stored Id {Id}", id);
```